qiskit/circuit/quantumcircuit.py

```diff
@@ -4454,7 +4451,7 @@ def prepare_state(
 
     def initialize(
         self,
-        params: Sequence[complex] | str | int,
+        params: Statevector | Sequence[complex] | str | int,
```

Comment:
`Sequence[complex]` is overspecified (my pet complaint that `Sequence` is pretty much _never_ correct as a type hint), but leaving aside `.index` and `.count`, `Statevector` would actually have already been covered by a type-hint of `numpy.typing.ArrayLike`, which is more appropriate.

That's not important, though, I just like complaining about `Sequence`.

Reply:
But this would be nice to rework consistently in all type hints (though `numpy.typing.ArrayLike` is a bit long 😛)